src/git.rs

```diff
@@ -0,0 +1,300 @@
+        SubgraphPushContextInput {
+            branch: self.branch,
+            commit: self.commit,
+            committer: self.committer,
+            remote_url: self.remote_url,
+            message: self.message,
+        }
+    }
+}
+
+type SubgraphCheckContextInput = subgraph::check::check_partial_schema_query::GitContextInput;
+impl Into<SubgraphCheckContextInput> for GitContext {
+    fn into(self) -> SubgraphCheckContextInput {
+        SubgraphCheckContextInput {
+            branch: self.branch,
+            commit: self.commit,
+            committer: self.committer,
+            remote_url: self.remote_url,
+            message: self.message,
+        }
+    }
+}
+
+impl GitContext {
+    pub fn new(
+        override_branch: Option<String>,
+        override_committer: Option<String>,
+        override_commit: Option<String>,
+        override_remote_url: Option<String>,
+    ) -> Self {
+        let git = git_info::get();
+
+        let branch = if override_branch.is_none() {
+            git.current_branch
+        } else {
+            override_branch
+        };
+        let commit = if override_commit.is_none() {
+            git.head.last_commit_hash_short
+        } else {
+            override_commit
+        };
+
+        let mut remote_url = override_remote_url;
+        let mut committer = override_committer;
+
+        // if both remote_url and committer have values, we don't need to
+        // worry about executing this block
+        if remote_url.is_none() || committer.is_none() {
+            if let Some(mut config) = git.config {
+                // use the local git remote url if not provided in env var
+                remote_url = if remote_url.is_none() {
+                    config.remove("remote.origin.url")
+                } else {
+                    remote_url
+                };
+
+                // if the committer from env is None, build from git info
+                committer = if committer.is_none() {
+                    let user = config.remove("user.name").unwrap_or_else(|| "".to_string());
+                    let email = config
+                        .remove("user.email")
+                        .unwrap_or_else(|| "".to_string());
+                    let email = if email.is_empty() {
+                        email
+                    } else {
+                        format!("<{}>", email)
+                    };
+
+                    // build final formatted committer
+                    if user.is_empty() && email.is_empty() {
+                        None
+                    } else {
+                        Some(format!("{} {}", user, email))
+                    }
+                } else {
+                    committer
+                };
+            };
+        };
+
+        let remote_url = if let Some(remote) = remote_url {
+            GitContext::sanitize_remote_url(&remote)
+        } else {
+            None
+        };
```

Comment:
```rust
let remote_url = remote_url.map(|url| GitContext::sanitize_remote_url(&url));
```

Reply:
The result of this would also be the wrong type, it'd be `Option<Option<String>>` instead of `Option<String>`